addons/web/static/src/core/notebook/notebook.js

```diff
@@ -57,14 +92,33 @@ export class Notebook extends Component {
         }
     }
 
+    computePages(props) {
+        if (!props.slots && !props.pages) {
+            return null;
+        }
+        const pages = [];
+        const pagesWithIndex = [];
+        Object.entries({ ...props.slots, ...props.pages }).forEach(([k, v]) => {
```

Comment:
i usually prefer `for` loop, so I would write this instead:
```js
for (let [k,v] in Object.entries({ ...props.slots, ...props.pages })) {
    ...
}
```
but it's more a question of taste